superset-frontend/src/SqlLab/components/SaveDatasetModal/index.tsx

```diff
@@ -158,49 +162,56 @@ export const SaveDatasetModal: FunctionComponent<SaveDatasetModalProps> = ({
 
     exploreChart({
       ...EXPLORE_CHART_DEFAULT,
-      datasource: `${datasetToOverwrite.datasetId}__table`,
+      datasource: `${datasetToOverwrite.datasetid}__table`,
       selected_columns: query.results.selected_columns,
     });
   };
 
-  const getUserDatasets = async (searchText = '') => {
-    // Making sure that autocomplete input has a value before rendering the dropdown
-    // Transforming the userDatasetsOwned data for SaveModalComponent)
-    const { userId } = user;
-    if (userId) {
-      const queryParams = rison.encode({
-        filters: [
-          {
-            col: 'table_name',
-            opr: 'ct',
-            value: searchText,
-          },
-          {
-            col: 'owners',
-            opr: 'rel_m_m',
-            value: userId,
-          },
-        ],
-        order_column: 'changed_on_delta_humanized',
-        order_direction: 'desc',
-      });
-
-      const response = await makeApi({
-        method: 'GET',
-        endpoint: '/api/v1/dataset',
-      })(`q=${queryParams}`);
-
-      return response.result.map(
-        (r: { table_name: string; id: number; owners: [DatasetOwner] }) => ({
-          value: r.table_name,
-          datasetId: r.id,
-          owners: r.owners,
-        }),
-      );
-    }
+  const loadDatasetOverwriteOptions = useCallback(
+    async (input = '') => {
+      const { userId } = user;
+      if (userId) {
```

Comment:
Looking at the `User` type, there's a comment that guest users don't have an `userId`. My question is, can guest users access this part of the application? It would be weird that a guest user is able to save a dataset. If they can't then we can assume that the `userId` always exists and remove this `if` statement.

Reply:
Oh great! I was just respecting as much logic from before when hitting the API. But let me confirm that and remove such `if` if that's the case. Thanks